include/ruby/internal/core/rbasic.h

```diff
@@ -153,4 +166,13 @@ RBASIC_CLASS(VALUE obj)
     return RBASIC(obj)->klass;
 }
 
+static inline void
+RBASIC_RESET_FLAGS(VALUE obj)
```

Comment:
Should this be in public headers (it is in the current state of this PR) or under `internal/**/*.h`?
IOW, do you think any C extension would need this, or just CRuby .c files?

Reply:
This is meant to be internal only.